src/main/java/microsoft/exchange/webservices/data/core/PropertyBag.java

```diff
@@ -758,7 +758,7 @@ public Object getObjectFromPropertyDefinition(PropertyDefinition propertyDefinit
         propertyDefinition, serviceExceptionOut);
     serviceException = serviceExceptionOut.getParam();
     if (serviceException == null) {
-      return propertyValue;
+      return (T) propertyValue;
```

Comment:
The cast here is only necessary because the propertyValue has been assigned to Object? Maybe you can fix this aswell? I have also spotted other occurrences of object in this class.